packages/auth/src/__tests__/AuthProvider.test.tsx

```diff
@@ -711,9 +711,7 @@ describe('Custom auth provider', () => {
       </AuthProvider>
     )
 
-    await waitFor(() => mockedForgotPassword.mock.calls.length === 1)
-
-    expect.assertions(1)
+    await waitFor(() => expect(mockedForgotPassword.mock.calls.length).toBe(1))
```

Comment:
@Tobbe could you look at this one real quick when you have the chance? Seems like it was failing cause `waitFor` needs whatever it is to throw, not just return `false`: https://testing-library.com/docs/dom-testing-library/api-async/#waitfor

Reply:
The `toBe(1)` change looks good. But I'd still like to keep the `expect.assertions` call to also be there, to make sure `expect(username).toEqual('username')` was executed

Reply:
I think I read your comment too quickly; I'm not sure if that code block would do what you want. It need a closer look

Reply:
@Tobbe after looking at it more, I don't think `expect.assertions` is going to work:

<img width="653" alt="image" src="https://user-images.githubusercontent.com/32992335/214148668-a368c0c5-563e-4ffd-8171-50c0cae88781.png">

`waitFor` executes the function it's given several times till it passes, so we can't rely on it calling the function a certain amount of times. 

I changed to the assertion to `toBeCalledWith` to test that `mockForgotPassword` is called with `"username"`. And won't the expect in `mockForgotPassword` fail if it's called with anything else?

`toBeCalledWith` will fail if it's never called at all, so it feels like all the bases are covered here, but I could ass `toHaveBeenCalled` for extra measure if you want.

Reply:
`toBeCalledWith` sounds like a good way to do it 👍